src/SIM/SIMbase.h

```diff
@@ -751,6 +754,28 @@ class SIMbase : public SIMinput, public SIMdependency
 private:
   size_t nIntGP; //!< Number of interior integration points in the whole model
   size_t nBouGP; //!< Number of boundary integration points in the whole model
+
+  //! \brief Class holding a MADOF for a given number of dofs on a given basis
+  class MADof {
+    public:
+      //! \brief Dummy constructor needed due to std::map.
+      MADof() {}
+      //! \brief Constructor.
+      //! \param[in] myModel The patch vector to setup the MADOF for
+      //! \param[in] nodes Total number of global nodes in model
+      //! \param[in] basis The basis to specify number of DOFs for
+      //! \param[in] nndof Number of DOFs on given basis
+      MADof(const PatchVec& myModel, size_t nodes,
+            unsigned char basis, unsigned char nndof);
+
+      //! \brief Access MADof array.
+      const std::vector<int>& get() const { return madof; }
+
+    protected:
+      std::vector<int> madof; //!< The MADOF array
+  };
+
+  mutable std::map<int, MADof> addMADOFs; //!< Additional MADOF arrays.
```

Comment:
Er det grunner jeg ikke ser enda til at du ønsker å ha en egen klasse for disse MADOF-ene, og ikke bare en std::map<int, std::vector<int> > + en private SIMbase-metode for å initialisere denne?